_rules/explicit-SVG-image-has-name-7d6734.md

```diff
@@ -0,0 +1,187 @@
+
+## Expectation
+
+Each target element has an [accessible name](#accessible-name) that is not only [whitespace](#whitespace).
+
+## Assumptions
+
+This rule assumes that the presence of one of the roles outlined in the applicability indicates the authors intent to include the element in the accessibility tree and thus convey information to the user about that element.
+
+## Accessibility Support
+
+The [HTML Accessibility API Mappings](https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings) specify that the `<svg>` element has an implicit role of `graphics-document`. However browser support for the `graphics-document` role and the [SVG Accessibility API Mappings (work in progress)](https://www.w3.org/TR/svg-aam-1.0) is inconsistent.
+
+This rule is limited to the explicit use of roles, as a clear indication that content should convey meaning, until the [SVG Accessibility API Mappings (work in progress)](https://www.w3.org/TR/svg-aam-1.0) is more stable and browser support is more consistent.
+
+Browser and assistive technology support for SVG `<title>` and `<desc>` elements is currently inconsistent. Using WAI ARIA in combination with the `img` role for non-decorative `<svg>` elements significantly improves accessibility browser support.
+
+Until browser support for the [SVG Accessibility API Mappings (work in progress)](https://www.w3.org/TR/svg-aam-1.0) is more consistent it is recommended to explicitly remove decorative <svg> elements from the accessibility tree.
+
+## Background
+
+- [Understanding Success Criterion 1.1.1: Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
+- [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0)
+- [WAI-ARIA Graphics Module](https://www.w3.org/TR/graphics-aria-1.0/)
+- [ARIA4: Using a WAI-ARIA role to expose the role of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4)
+- [ARIA6: Using aria-label to provide labels for objects](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6)
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `<svg>` element has an explicit role of `img` and an accessible name from the `<title>` element that does not contain only whitespace.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<title>1 circle</title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Passed Example 2
+
+The `<circle>` element has an explicit role of `graphics-symbol` and an accessible name from the `aria-label` attribute that does not contain only whitespace.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="https://www.w3.org/2000/svg" role="img">
```

Comment:
Ran this by some folks and it seems suspect. If the SVG is taking on a role of img then the accessible name should be calculated as though it is an img element. If that is the case then using enclosed content for the accessible name is now excluded. So I would expect this to fail for lack of an accessible name. In other words, the circle element within might have an accessible name, but it doesn't count because the SVG is now an img.

Reply:
I have removed the `img` role from the `svg` element. The `circle` element is still applicable and has an accessible name.

Reply:
This has the expected behaviour in all my test 😄